qadence/backends/gpsr.py

```diff
@@ -48,7 +48,13 @@ def single_gap_psr(
     Returns:
         Tensor: tensor containing derivative values
     """
-
+    device = torch.device("cpu")
+    try:
+        device = [v.device for v in param_dict.values()][0]
+    except Exception:
+        pass
```

Comment:
I would add some explicit message here

Reply:
its just a hotfix, we will move this logic to pyq soon